### PR TITLE
Add Next.js Zip example

### PR DESCRIPTION
**Pull request by julianbonilla**
*Issue #, if available:*

*Description of changes:*

Add an example for Next.js 12 using Zip package type.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: examples/nextjs-zip/app/run.sh [hunk 1]
```diff
@@ -0,0 +1,3 @@
+#!/bin/bash
+
+node server.js
```
Comment (bnusunny) [inline examples/nextjs-zip/app/run.sh:3]: Two updates for this line: 

1. `server.js` is in `standalone` directory. 2. 
2. Add `exec` will allow the nodejs process to receive SIGTERM for graceful shutdown. 

`exec node standalone/server.js`

**Review by bnusunny — CHANGES_REQUESTED** [review]

File: examples/nextjs-zip/app/run.sh (hunk 1 above)
Comment (julianbonilla) [reply examples/nextjs-zip/app/run.sh]: `server.js` is at the root of the artifacts directory, so I updated to `exec node server.js`

```bash
$ pwd
aws-lambda-web-adapter/examples/nextjs-zip/.aws-sam/build/NextjsFunction
$ tree -I node_modules/
.
├── .next
├── node_modules
├── package.json
├── public
│   ├── favicon.ico
│   └── vercel.svg
├── run.sh
└── server.js
```

Comment (bnusunny) [reply examples/nextjs-zip/app/run.sh]: I built this example. The server.js was under `standalone` folder. If this is not intended, the build command in Makefile should be checked. 

[ec2-user@ip-172-31-42-124 nextjs-zip]$ ls -l .aws-sam/build/NextjsFunction/
total 4
-rw-rw-r-- 1 ec2-user ec2-user 44 Sep  2 15:28 run.sh
drwxrwxr-x 5 ec2-user ec2-user 90 Sep  2 15:28 standalone

Comment (julianbonilla) [reply examples/nextjs-zip/app/run.sh]: Hmm, I can't reproduce.  You're running `sam build` from `nextjs-zip` directory?

Comment (bnusunny) [reply examples/nextjs-zip/app/run.sh]: Yes. I run `sam build` from `nextjs-zip` directory on an ec2 instance. The OS is Amazon Linux 2. The shell is bash.

File: examples/nextjs-zip/app/Makefile [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+build:
+	npm run build
+
+artifacts:
+	# Static resources ideally served from CloudFront
+	cp -r public/ .next/standalone/public
+	cp -r .next/static/ .next/standalone/.next/static
+	
+	# Copy artifacts for deployment
+	cp -r .next/standalone/ $(ARTIFACTS_DIR)
```
Comment (bnusunny) [inline examples/nextjs-zip/app/Makefile:13]: Change this line to this: `cp -r .next/standalone/. $(ARTIFACTS_DIR)`

This works on both LInux and Macos.

**Review by bnusunny — CHANGES_REQUESTED** [review]

File: examples/nextjs-zip/app/Makefile (hunk 2 above)
Comment (bnusunny) [reply examples/nextjs-zip/app/Makefile:13]: Please also look at other copy commands in the Makefile.

**Review by bnusunny — APPROVED** [review]
LGTM

**Comment by bnusunny** [comment]
@julianbonilla  This is a great example. Thanks for the contribution!

**Pull request merged**